### DISCUSSION TITLE
Idea/QOL-Enhancement: Dragging properties from inspector into code without "Quotation marks"

### DISCUSSION THREAD

**Discussion by nosdregamon** [godotengine/godot-proposals#8012] [Editor]
I just noticed, that I can drag properties from the inspector into code. Just search for the the property I want, say Node2d->Position, drag the name and I got "position" in my script - that's awesome (and I feel silly for not trying that earlier).

However, this always ends up "quoted".  However, they are always in quotation marks, so it's _**"position"**_ not just _**position**_ and in scripts in need the latter. So I remove the quotation marks by hand. Would be a small but "nice to have" improvement to implement an option to drag without quotes. 

Ideally it would be it's own modifier shortcut (say "CTRL+Drag&Drop" to get the desired result). But currently nothing in that direction exists.

Someone more savvy in the Godot sources pointed me to...

https://github.com/godotengine/godot/blob/6916349697a4339216469e9bf5899b983d78db07/editor/plugins/script_text_editor.cpp#L1917-L1927

... which boils down to: it's currently always quoting.

I think the addition would be minimal such as replacing 'const String text_to_drop = _quote_drop_data(String(d["property"]));' with 
      const String text_to_drop = drop_modifier_pressed ? String(d["property"]) : _quote_drop_data(String(d["property"]));

But it looks pretty ugly, and I can't get Godot to compile on my cheap notebook.

**Comment by theraot** [comment]
You seem to have everything you need to write a proper proposal.

Addendum: I think I can do that change, the only thing giving me pause is considering if CTRL could do something else.

**Reply by theraot** [reply]
I'm linking here your follow up proposal: https://github.com/godotengine/godot-proposals/issues/8014
And my counter-proposal: https://github.com/godotengine/godot-proposals/issues/8017
